docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,595 @@
+    ![Breakpoint set next to the first line in the add_item function.](images/fastapi-tutorial/debugger_breakpoint.png)
+
+5. Start the debugger (`kb(workbench.action.debug.start)`), and then navigate to http://127.0.0.1:8000/docs in the browser.
+
+    You'll see a Swagger interface with the two endpoints available in the app: `/items` and root (`/`).
+    ![Swagger UI displaying two endpoints: /items and /](images/fastapi-tutorial/fastapi_first_swagger_page.png)
+
+6. Click on the down arrow next to the `/items` route to expand it, and click on the "Try it out" button that appears on the right side.
+    ![Try it out button displayed next to the /items route in the Swagger UI.](images/fastapi-tutorial/fastapi_tryitout_button.png)
+
+7. Add a grocery list item by passing a string to the "item" field, and a number to "quantity". For example, you could provide `{"item" : "apple", "quantity": 2 }`.
+
+8. Click on the "Execute" button.
+    ![Execute button displayed below the /items route.](images/fastapi-tutorial/fastapi_execute_button.png)
+
+9. Open VS Code again and notice the debugger has stopped at the breakpoint we set earlier.
+    ![Debugger stopped at the breakpoint set in the add_item function.](images/fastapi-tutorial/fastapi_breakpoint_hit.png)
+
+On the left side, you will see all local and global variables defined at this point in the Variables window, under the **Run and Debug** view:
+
+![Variables window displayed in the Run and Debug view, with the item and grocery_list variables highlighted.](images/fastapi-tutorial/fastapi_debugger_variables.png)
+
+Now let’s leverage VS Code's Debug Console to do some exploration.
+
+10.	Select the `quantity <= 0` statement, right-click on the editor and select **Evaluate in Debug Console**:
+    ![Evaluate in Debug Console option displayed in the context menu when right-clicking on a line of code.](images/fastapi-tutorial/fastapi_evaluate_debug_console.png)
+
+    This will open the Debug Console and run the selected expression. As expected, the expression will be evaluated to `False`.
+
+The Debug Console can be a powerful tool to quickly test expressions and better understand the state of your code at the time of the breakpoint. You can also use it to run arbitrary code, such as calling functions or printing variables. Learn more about Python debugging in VS Code in the [Python tutorial](python-tutorial.md#configure-and-run-the-debugger).
+
+
+Finally, let's add the remaining routes for the application so we can list all items or specific items, as well as remove them from our grocery list. You can leave the debugger running as it will automatically reload the application when you save the changes you'll make in the next step.
```

Comment:
Should we have the user remove the breakpoint? When I replaced the code in main.py the breakpoint no longer aligned with the if statement.